### ISSUE TITLE
internal compiler error: no warnings or errors encountered even though `delayed_good_path_bugs` issued

### ISSUE THREAD

**Issue by djeedai** [rust-lang/rust#102572]
### Code

This triggers on CI on Linux; I don't have a Linux to try a repro:

```
git clone https://github.com/djeedai/bevy_hanabi.git
git checkout 44ed04bd77d2a9eab8622b6c01d9cb906980bc49
cargo test --no-default-features --features 2d
```

### Meta

https://github.com/djeedai/bevy_hanabi/actions/runs/3168209195/jobs/5159251886#step:4:47
```
 /home/runner/.cargo/bin/rustc -V
rustc 1.66.0-nightly (57f097ea2 2022-10-01)
/home/runner/.cargo/bin/cargo -V
cargo 1.66.0-nightly (f5fed93ba 2022-09-27)
/home/runner/.cargo/bin/rustup -V
rustup 1.25.1 (bb60b1e89 2022-07-12)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.66.0-nightly (57f097ea2 2022-10-01)`
```

https://github.com/djeedai/bevy_hanabi/actions/runs/3168209195/jobs/5159251886#step:7:311
```
note: rustc 1.66.0-nightly (57f097ea2 2022-10-01) running on x86_64-unknown-linux-gnu
note: compiler flags: --crate-type lib -C embed-bitcode=no -C debuginfo=2
```

### Error output

https://github.com/djeedai/bevy_hanabi/actions/runs/3168209195/jobs/5159251886

```
error: internal compiler error: no warnings or errors encountered even though `delayed_good_path_bugs` issued
error: internal compiler error: trimmed_def_paths constructed
```

<details><summary><strong>Backtrace</strong></summary>
<p>

```
thread 'rustc' panicked at 'Box<dyn Any>', compiler/rustc_errors/src/lib.rs:1536:13
stack backtrace:
   0:     0x7f676f414380 - std::backtrace_rs::backtrace::libunwind::trace::h80b67585f93dcdfa
                               at /rustc/57f097ea25f2c05f424fc9b9dc50dbd6d399845c/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   1:     0x7f676f414380 - std::backtrace_rs::backtrace::trace_unsynchronized::h1727d1d942e500b9
                               at /rustc/57f097ea25f2c05f424fc9b9dc50dbd6d399845c/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f676f414380 - std::sys_common::backtrace::_print_fmt::h1d58d8daeea91706
                               at /rustc/57f097ea25f2c05f424fc9b9dc50dbd6d399845c/library/std/src/sys_common/backtrace.rs:66:5
   3:     0x7f676f414380 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h813c9b7ed78580ba
                               at /rustc/57f097ea25f2c05f424fc9b9dc50dbd6d399845c/library/std/src/sys_common/backtrace.rs:45:22
   4:     0x7f676f46f14e - core::fmt::write::h3eecc0c99e4c5326
                               at /rustc/57f097ea25f2c05f424fc9b9dc50dbd6d399845c/library/core/src/fmt/mod.rs:1209:17
   5:     0x7f676f404f05 - std::io::Write::write_fmt::h169b89be5a4eb2e3
                               at /rustc/57f097ea25f2c05f424fc9b9dc50dbd6d399845c/library/std/src/io/mod.rs:1679:15
   6:     0x7f676f417123 - std::sys_common::backtrace::_print::h8470ccc2c46bc222
                               at /rustc/57f097ea25f2c05f424fc9b9dc50dbd6d399845c/library/std/src/sys_common/backtrace.rs:48:5
   7:     0x7f676f417123 - std::sys_common::backtrace::print::h59bfcbb529aad43f
                               at /rustc/57f097ea25f2c05f424fc9b9dc50dbd6d399845c/library/std/src/sys_common/backtrace.rs:35:9
   8:     0x7f676f417123 - std::panicking::default_hook::{{closure}}::h19081fa8b049df20
                               at /rustc/57f097ea25f2c05f424fc9b9dc50dbd6d399845c/library/std/src/panicking.rs:267:22
   9:     0x7f676f416dfa - std::panicking::default_hook::h8f0eea70584d6151
                               at /rustc/57f097ea25f2c05f424fc9b9dc50dbd6d399845c/library/std/src/panicking.rs:286:9
  10:     0x7f6771c83c21 - <rustc_driver[5f28de9a0492c992]::DEFAULT_HOOK::{closure#0}::{closure#0} as core[2f8af133219d6c11]::ops::function::FnOnce<(&core[2f8af133219d6c11]::panic::panic_info::PanicInfo,)>>::call_once::{shim:vtable#0}
  11:     0x7f676f41795b - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::hb9a306ebd1946910
                               at /rustc/57f097ea25f2c05f424fc9b9dc50dbd6d399845c/library/alloc/src/boxed.rs:1952:9
  12:     0x7f676f41795b - std::panicking::rust_panic_with_hook::h6b2b2d988170cf5c
                               at /rustc/57f097ea25f2c05f424fc9b9dc50dbd6d399845c/library/std/src/panicking.rs:673:13
  13:     0x7f6772dcbfb1 - std[281da818632cbbdf]::panicking::begin_panic::<rustc_errors[1020869c0a53ac34]::ExplicitBug>::{closure#0}
  14:     0x7f6772dca8c6 - std[281da818632cbbdf]::sys_common::backtrace::__rust_end_short_backtrace::<std[281da818632cbbdf]::panicking::begin_panic<rustc_errors[1020869c0a53ac34]::ExplicitBug>::{closure#0}, !>
  15:     0x7f6772dc4416 - std[281da818632cbbdf]::panicking::begin_panic::<rustc_errors[1020869c0a53ac34]::ExplicitBug>
  16:     0x7f6772dc6b66 - std[281da818632cbbdf]::panic::panic_any::<rustc_errors[1020869c0a53ac34]::ExplicitBug>
  17:     0x7f67717d7ff3 - <rustc_errors[1020869c0a53ac34]::HandlerInner as core[2f8af133219d6c11]::ops::drop::Drop>::drop
  18:     0x7f67709b9448 - core[2f8af133219d6c11]::ptr::drop_in_place::<rustc_session[786cb1c5da006f9b]::parse::ParseSess>
  19:     0x7f6770978ba8 - <alloc[ab68081038e0c5ba]::rc::Rc<rustc_session[786cb1c5da006f9b]::session::Session> as core[2f8af133219d6c11]::ops::drop::Drop>::drop
  20:     0x7f6770975aed - core[2f8af133219d6c11]::ptr::drop_in_place::<rustc_interface[a2e36ac52ab23c81]::interface::Compiler>
  21:     0x7f677097348c - rustc_interface[a2e36ac52ab23c81]::interface::create_compiler_and_run::<core[2f8af133219d6c11]::result::Result<(), rustc_errors[1020869c0a53ac34]::ErrorGuaranteed>, rustc_driver[5f28de9a0492c992]::run_compiler::{closure#1}>
  22:     0x7f6770971982 - <scoped_tls[430d84ded6e482c8]::ScopedKey<rustc_span[efd8b0f1895919e0]::SessionGlobals>>::set::<rustc_interface[a2e36ac52ab23c81]::interface::run_compiler<core[2f8af133219d6c11]::result::Result<(), rustc_errors[1020869c0a53ac34]::ErrorGuaranteed>, rustc_driver[5f28de9a0492c992]::run_compiler::{closure#1}>::{closure#0}, core[2f8af133219d6c11]::result::Result<(), rustc_errors[1020869c0a53ac34]::ErrorGuaranteed>>
  23:     0x7f677097166f - std[281da818632cbbdf]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[a2e36ac52ab23c81]::util::run_in_thread_pool_with_globals<rustc_interface[a2e36ac52ab23c81]::interface::run_compiler<core[2f8af133219d6c11]::result::Result<(), rustc_errors[1020869c0a53ac34]::ErrorGuaranteed>, rustc_driver[5f28de9a0492c992]::run_compiler::{closure#1}>::{closure#0}, core[2f8af133219d6c11]::result::Result<(), rustc_errors[1020869c0a53ac34]::ErrorGuaranteed>>::{closure#0}, core[2f8af133219d6c11]::result::Result<(), rustc_errors[1020869c0a53ac34]::ErrorGuaranteed>>
  24:     0x7f6771ae5550 - <<std[281da818632cbbdf]::thread::Builder>::spawn_unchecked_<rustc_interface[a2e36ac52ab23c81]::util::run_in_thread_pool_with_globals<rustc_interface[a2e36ac52ab23c81]::interface::run_compiler<core[2f8af133219d6c11]::result::Result<(), rustc_errors[1020869c0a53ac34]::ErrorGuaranteed>, rustc_driver[5f28de9a0492c992]::run_compiler::{closure#1}>::{closure#0}, core[2f8af133219d6c11]::result::Result<(), rustc_errors[1020869c0a53ac34]::ErrorGuaranteed>>::{closure#0}, core[2f8af133219d6c11]::result::Result<(), rustc_errors[1020869c0a53ac34]::ErrorGuaranteed>>::{closure#1} as core[2f8af133219d6c11]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  25:     0x7f676f421433 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h26c13cce480449d5
                               at /rustc/57f097ea25f2c05f424fc9b9dc50dbd6d399845c/library/alloc/src/boxed.rs:1938:9
  26:     0x7f676f421433 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h1d016e890f9a0fa4
                               at /rustc/57f097ea25f2c05f424fc9b9dc50dbd6d399845c/library/alloc/src/boxed.rs:1938:9
  27:     0x7f676f421433 - std::sys::unix::thread::Thread::new::thread_start::ha5f4632b746acf38
                               at /rustc/57f097ea25f2c05f424fc9b9dc50dbd6d399845c/library/std/src/sys/unix/thread.rs:108:17
  28:     0x7f676f2e8609 - start_thread
  29:     0x7f676f20b133 - clone
  30:                0x0 - <unknown>
```

</p>
</details>

**Comment by djeedai** [comment]
Possibly a duplicate of #102561

**Comment by compiler-errors** [comment]
Yeah, it's a duplicate of #102561. I won't bother minimizing a test for this because the stack trace looks identical.

**Comment by djeedai** [comment]
Closing as duplicate of #102561.
